docs/src/pages/docs/crocks/Result.md

```diff
@@ -0,0 +1,1566 @@
+import map from 'crocks/pointfree/map'
+import maybeToEither from 'crocks/Result/maybeToEither'
+import merge from 'crocks/Pair/merge'
+import objOf from 'crocks/helpers/objOf'
+import prop from 'crocks/Maybe/prop'
+import safeLift from 'crocks/Maybe/safeLift'
+
+const { Left, Right } = Either
+const { Ok } = Result
+
+eitherToResult(Left('no good'))
+//=> Err "no good"
+
+eitherToResult(Right('so good'))
+//=> Ok "so good"
+
+// safeInc :: a -> Maybe Number
+const safeInc =
+  safeLift(isNumber, x => x + 1)
+
+// incProp :: String -> a -> Maybe Number
+const incProp = key =>
+  composeK(safeInc, prop(key))
+
+// incResult :: String -> a -> Result [ String ] Object
+const incResult = key => maybeToEither(
+  [ `${key} is not valid` ],
+  composeB(
+    map(objOf(key)),
+    incProp(key)
+  )
+)
+
+// incThem :: a -> Result [ String ] Object
+const incThem = composeB(
+  merge(liftA2(assign)),
+  fanout(incResult('b'), incResult('a'))
+)
+
+Result.of({})
+  .chain(eitherToResult(incThem))
+//=> Err [ "b is not valid", "a is not valid" ]
+
+Result.of({ a: 33 })
+  .chain(eitherToResult(incThem))
+//=> Err [ "b is not valid" ]
+
+Result.of({ a: 99, b: '41' })
+  .chain(eitherToResult(incThem))
+//=> Err [ "b is not valid" ]
+
+Result.of({ a: 99, b: 41 })
+  .chain(eitherToResult(incThem))
+//=> Ok { a: 100, b: 42 }
+
+Ok(Left('Err'))
+  .chain(eitherToResult)
+//=> Err "Err"
+
+Ok(Right('42'))
+  .chain(eitherToResult)
+//=> Ok "42"
+```
+
+#### firstToResult
+
+`crocks/Result/firstToResult`
+
+```haskell
+firstToResult :: e -> First a -> Result e a
+firstToResult :: e -> (a -> First b) -> a -> Result e b
+```
+
+Used to transform a given [`First`][first] instance to a `Result`
+instance or flatten a `Result` of [`First`][first] into a `Result` when chained, 
+`firstToMaybe` will turn a non-empty instance into an [`Ok`](#ok) wrapping
+the original value contained within the [`First`][first]. All empty instances
+will map to an [`Err`](#err) with the given value.
+
+Like all `crocks` transformation functions, `firstToMaybe` has two possible
+signatures and will behave differently when passed either
+a [`First`][first] instance or a function that returns an instance
+of [`First`][first]. When passed the instance, a transformed `Result` is
+returned. When passed a [`First`][first] returning function, a function will be
+returned that takes a given value and returns a `Result`.
+
+```javascript
+import First from 'crocks/First'
+
+import composeB from 'crocks/combinators/composeB'
+import concat from 'crocks/pointfree/concat'
+import firstToResult from 'crocks/Result/firstToResult'
+import flip from 'crocks/combinators/flip'
+import mapReduce from 'crocks/helpers/mapReduce'
+import prop from 'crocks/Maybe/prop'
+
+const { empty } = First
+
+// Person :: { name: String, age: Number }
+
+// createPerson :: (String, Number) -> Person
+const createPerson = (name, age) => ({
+  name, age
+})
+
+// liftName :: Person -> First (Maybe String)
```

Comment:
As all `First` instances wrap a `Maybe` it is not needed in the siggy. this can just be
```js
// liftName :: Person -> First String
```

Reply:
Good to know